byos/app-modernization-no-sql/deploy/deployAll.ps1

```diff
@@ -71,14 +69,21 @@ for ($i = 1; $i -le $teamCount; $i++)
         else
         {
             #report error for incorrect RG deployment
-            Write-Output("Deployment failed for team: " + $teamName + ". Resource Groups could not be found.");
+            $failedEnvironments += 1
+            Write-Error("Deployment failed for team: " + $teamName + ". Resource Groups could not be found.");
         }
     }
     catch {
         #report error, team deployment failure
-        Write-Output "An error was encountered, script could not complete:  $($PSItem.ToString())";
+        $failedEnvironments += 1
+        Write-Error "An error was encountered, script could not complete:  $($PSItem.ToString())";
     }
 }
 
 #report operation completed
-Write-Output "All resources are deployed.  Enjoy the OpenHack!";
+if($failedEnvironments -eq 0) {
+    Write-Output "All resources are deployed.  Enjoy the OpenHack!";
+}
+else {
+    Write-Error "$failedEnvironments environments failed either provisioning or validation. Please check output for more details"
```

Comment:
One thing that I considered here was to in the error path add `exit 1` to provide an opportunity for automated execution to see the execution as unsuccessful and bubble to relevant reporting. I think that's an appropriate change, but couldn't be sure of ramifications to existing processes, so I didn't include it in this round.